src/node/mod.rs

```diff
@@ -380,3 +380,21 @@ pub fn dev_path(dev: dev_t, ty: NodeType) -> io::Result<PathBuf> {
         ),
     ))
 }
+
+/// Returns a Vector with all DRM Nodes we managed to find. There might be duplicates.
+pub fn drm_get_devices() -> Result<Vec<DrmNode>, io::Error> {
+    let mut devices: Vec<DrmNode> = vec![];
+
+    fs::read_dir("/dev")?
+        .for_each(|entry| {
```

Comment:
How about making this a `.map().collect()` rather than a side-effecting `for_each()`? If that, use regular control flow with `for entry in fs::read_dir(..)? { .. }`.

The `.collect()` can be dropped if using @notgull's suggestion to return an iterator instead.

Reply:
Wouldn't we need to filter the results in some way if we use `.map()`? Asking as we check all devices and likely a lot of them aren't DRM devices, so we need to skip them. This validation is done right now by calling `DrmNode::from_path()`.

Reply:
Such filtering can be implemented by a [`.filter_map()`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter_map), but I'm hesitant to `stat()` every device on the system to see if it is a DRM device. 

Since `drmGetDevices2()` was requested, I expected this feature to be built after that function. While it does also [appear to ignore any error from processing individual devices](https://gitlab.freedesktop.org/mesa/drm/-/blob/e68e9b8013f0cb5630afd156d330dc9665d9d98f/xf86drm.c#L4789-4791), this function [does open `DRM_DIR_NAME`](https://gitlab.freedesktop.org/mesa/drm/-/blob/e68e9b8013f0cb5630afd156d330dc9665d9d98f/xf86drm.c#L4783) which is [`/dev/dri` on every platform except FreeBSD where it is `/dev`](https://gitlab.freedesktop.org/mesa/drm/-/blob/e68e9b8013f0cb5630afd156d330dc9665d9d98f/xf86drm.h#L79-90). This way we don't have to `stat()` devices that we know won't be DRM devices in the first place.

Reply:
Hey thanks for the pointers. I somehow completely missed the fact that `DRM_DIR_NAME` was different in certain platforms. I will update the code to also use `/dev/dri` for all platforms except FreeBSD.